### DISCUSSION TITLE
Chat platform preference – Discord, gitter, something else?

### DISCUSSION THREAD

**Discussion by raquo** [raquo/Laminar#114] [General]
Laminar chat is currently hosted on Gitter, mostly because that was the platform of choice for Scala projects at the time. Many Scala communities seem to be moving to Discord. Gitter seems to be on life support. It doesn't look like anyone is working on it anymore.

I'm thinking if we should stay on Gitter, or switch to Discord, or something else. My basic requirements:
* Must be hosted in the cloud (no maintenance required)
* Must be free, or at worst not more than $50/year
* Must have decent search
* The entire message history must be accessible and searchable with no limits

Please vote on your preferred platform, and leave a comment below. Additional questions:
* Do you often use **gitter** search to find answers to Laminar questions? Do you usually find them?
* Do you often use **web** search (e.g. on google) to find answers to Laminar questions, and end up finding answers on our gitter?

# Update – we're moving to Discord!

# 👉 https://discord.gg/JTrUxhq7sj

**Comment by raquo** [comment]
Upvote this if you'd rather stay on Gitter

**Reply by objektwerks** [reply]
I've always preferred Gitter over Discord, principally because Gitter is publicly searchable.

**Comment by raquo** [comment]
Upvote this if you'd rather switch to Discord

**Comment by raquo** [comment]
Upvote this to switch to [Zulip](https://zulip.com/) (assuming they approve us for the free open source plan).

I haven't looked too much into it, but it has some interesting features, and unlike Discord, zulip is open source, so we could potentially switch to self hosting it later if their free offering eventually goes away.

**Reply by raquo** [reply]
Zulip's own community chatroom, for reference: https://chat.zulip.org/

You can see that this product is quite different from standard chat. The main view is more like a forum. And if you navigate into one of the "streams" (e.g. [backend](https://chat.zulip.org/#narrow/stream/3-backend)), you will see the latest messages in that stream, grouped by thread. Pretty unconventional, but potentially useful, especially for higher trafficked communities (which we are not). I haven't made up my mind about this format yet.

One of the complaints I've heard about this thread-focused format is the lack of structure-less chat, which is sometimes desired. I heard that people solve this problem by creating an evergreen "general chat" thread.

**Comment by yurique** [comment]
Yeah, after getting acquired by Matrix gitter just stopped evolving. I've been using the `Element` (Matrix client) to chat in gitter, but it's not perfect either.

I don't think I ever managed to find anything using gitter search. I have, however, been able to **google** things up in the Laminar channel, sometimes from years ago. That is definitely an advantage of gitter. (I just wish they haven't ruined it :'( ).

I voted discord mostly because many other projects are already there. It's good UI-wise (feels smooth and), the built-in search is somewhat usable and useful – although no google indexing. On the other hand, I find it challenging to keep up in busy servers with lots of channels.

Zulip looks interesting. But I'd have to actively use it for a few weeks with a larger group of people to tell if it's any good (I would definitely like to see it in action as a replacement for Slack at work, but that's never happening =) ).

**Comment by yurique** [comment]
I know I voted for Discord already, but I've been checking out Zulip a little bit. I learned that (some) Rust teams are using it, and there's `rust-lang` community server – so Zulip must be working well enough.

The more I think about their approach with streams/topics, the more I like it – even for a rather small chat like we have it feels like it would be beneficial. Like, someone has a question or an idea – that's new topic, can be small or large, but focused on the question. Perfect, right? :)

**Reply by yurique** [reply]
Search seems to be quite functional (haven't challenged it with something hard, though)
![image](https://user-images.githubusercontent.com/919230/152651644-a590cea1-820d-4efb-8d36-cb9d34051a1a.png)

**Reply by yurique** [reply]
Wonder why Scala chats picked Discord 🤔

**Reply by objektwerks** [reply]
Gitter or Zulip works for me. Discord, though, has a shady past on privacy. Yet the Scala herd has moved to Discord. What's that about herds and cliffs?:)

**Reply by raquo** [reply]
Well Discord has the advantage of being popular. For us there are two concerns:
- Will the people who **ask** questions be willing to sign up for zulip?
- Will the people who **answer** questions be willing to keep a zulip tab open?

Personally I think it's not unfair to ask someone to take a minute to create a free account to get their question answered, and if they don't want to do that, they still have github discussions where they already have an account. So the question is mostly, will there be enough people monitoring the channel given the extra friction of using a chat platform that they don't use for anything else? And will this extra friction prevent some people from starting to answer questions.

What I mean is, I used to answer some questions in the scala-js gitter, just because I have gitter always open. But now that it's in Discord, I don't, because I don't routinely have Discord open. But for most people the situation would be reversed, they would be more likely to have Discord open. Of course I personally will have the Laminar chat open all the time anyway regardless of the platform, but if I'm the only one in there it'll get kinda lonely.
